cmd/boulder-migrator/main.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// For the applied DATETIME field to work properly with MySQL the DSN parameter
+// "parseTime=true" is required (e.g. "user:pass@tcp(localhost:3306)/boulder?parseTime=true")
+// otherwise sql.Row.Scan will throw an error ("[]uint8 -> time.Time is bad" or something).
+
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+	_ "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/go-sql-driver/mysql"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type Migration struct {
+	Id      int
+	Desc    string   // Basic explanation of what the migration does
+	UpSQL   []string
+	DownSQL []string
+}
+
+func getYN(question string) (answer bool) {
+	fmt.Printf("\n%s? [y/n]", question)
+	var free bool
+	for {
+		var resp string
+		fmt.Printf("\n# ")
+		_, err := fmt.Scanf("%s", &resp)
+		if err != nil {
+			fmt.Printf("%s: Please enter y or n\n", err)
+			continue
+		}
+		switch resp {
+		case "yes":
+			fallthrough
+		case "YES":
+			fallthrough
+		case "Yes":
+			fallthrough
+		case "Y":
+			fallthrough
+		case "y":
+			answer = true
+			free = true
+		case "no":
+			fallthrough
+		case "NO":
+			fallthrough
+		case "No":
+			fallthrough
+		case "N":
+			fallthrough
+		case "n":
+			answer = false
+			free = true
+		default:
+			fmt.Println("Please enter y or n")
+		}
+		if free {
+			break
+		}
+	}
+	return
+}
+
+func createMigrationTable(Sql *sql.DB) (err error) {
+	tx, err := Sql.Begin()
+	if err != nil {
+		return
+	}
+	_, err = tx.Exec("CREATE TABLE IF NOT EXISTS migrations (id INTEGER, description TEXT, applied DATETIME, down TEXT)")
```

Comment:
You should probably make id a UNIQUE PRIMARY. I think it's probably best to make it a string rather than int, just to simplify using datestamps pulled from filenames.